File: app/src/test/java/fr/free/nrw/commons/NearbyControllerTest.java
package fr.free.nrw.commons;

import org.junit.Test;
import org.junit.runner.RunWith;
import org.robolectric.RobolectricTestRunner;
import org.robolectric.RuntimeEnvironment;
import org.robolectric.annotation.Config;

import java.util.ArrayList;
import java.util.List;

import fr.free.nrw.commons.location.LatLng;
import fr.free.nrw.commons.nearby.NearbyBaseMarker;
import fr.free.nrw.commons.nearby.Place;

import static fr.free.nrw.commons.nearby.NearbyController.loadAttractionsFromLocationToBaseMarkerOptions;
import static org.hamcrest.CoreMatchers.is;
import static org.junit.Assert.assertThat;

@RunWith(RobolectricTestRunner.class)
@Config(constants = BuildConfig.class, sdk = 21, application = TestCommonsApplication.class)
public class NearbyControllerTest {

    @Test
    public void testNullAttractions() {
        LatLng location = new LatLng(0, 0, 0);

        List<NearbyBaseMarker> options = loadAttractionsFromLocationToBaseMarkerOptions(
                location, null, RuntimeEnvironment.application);

        assertThat(options.size(), is(0));
    }

    @Test
    public void testEmptyList() {
        LatLng location = new LatLng(0, 0, 0);
        List<Place> emptyList = new ArrayList<>();

        List<NearbyBaseMarker> options = loadAttractionsFromLocationToBaseMarkerOptions(
                location, emptyList, RuntimeEnvironment.application);

        assertThat(options.size(), is(0));
    }
}

File: app/src/test/kotlin/fr/free/nrw/commons/NearbyControllerTest.kt
package fr.free.nrw.commons

import fr.free.nrw.commons.location.LatLng
import fr.free.nrw.commons.nearby.NearbyController.loadAttractionsFromLocationToBaseMarkerOptions
import androidx.test.core.app.ApplicationProvider
import org.junit.Assert.assertEquals
import org.junit.Test
import org.junit.runner.RunWith
import org.robolectric.RobolectricTestRunner
import org.robolectric.annotation.Config

@RunWith(RobolectricTestRunner::class)
@Config(sdk = [21], application = TestCommonsApplication::class)
class NearbyControllerTest {

    @Test
    fun testNullAttractions() {
        val location = LatLng(0.0, 0.0, 0f)

        val options = loadAttractionsFromLocationToBaseMarkerOptions(
                location, null)

        assertEquals(0, options.size.toLong())
    }

    @Test
    fun testEmptyList() {
        val location = LatLng(0.0, 0.0, 0f)

        val options = loadAttractionsFromLocationToBaseMarkerOptions(
                location, emptyList())

        assertEquals(0, options.size.toLong())
    }
}
